src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part1/index.md

```diff
@@ -0,0 +1,430 @@
+
+#### Snaptrim + Recovery Effect on Client Latency
+
+To understand the effect of snaptrim and recovery ops on client latency, the plots shown below overlays the client latency profile against the snaptrim and recovery rate during a test run.
+
+[![](images/Img16_WPQ_Rec_vs_Snaptrim_vs_Avg_Clat.png)](images/Img16_WPQ_Rec_vs_Snaptrim_vs_Avg_Clat.png)
+
+<center>Fig. 16: WPQ - Avg Client Latency Variations During Recovery+Snaptrim Ops</center><br>
+
+The plots shown below provide an understanding of how the client latency varies during the snaptrim and recovery phase between the WPQ scheduler and the mClock profiles. It can easily be discerned that with the *high_client_ops*, the average client latency is on a slightly higher side until the snaptrim ops complete. It is evident that the snaptrim rate curve with the *high_client_ops* profile is steeper than the rate with the WPQ scheduler and this is the reason for the marginally high (8.6%) average client latency. But on the other hand, the average snaptrim rate is faster by around 18.7%.
+
+For the *balanced* and *high_recovery_ops* profiles, the recoveries can be seen to complete faster with a corresponding impact to the snaptrim and client ops. This is due to the nature of the QoS reservation and limit allocation to the client and best effort op classes in these profiles.
+
+[![](images/Img17_high_client_Profile_Rec_vs_Snaptrim_vs_Avg_Clat.png)](images/Img17_high_client_Profile_Rec_vs_Snaptrim_vs_Avg_Clat.png)
+
+<center>Fig. 17: high_client_ops - Avg Client Latency Variations During Recovery+Snaptrim Ops</center><br>
+
+[![](images/Img18_balanced_Profile_Rec_vs_Snaptrim_vs_Avg_Clat.png)](images/Img18_balanced_Profile_Rec_vs_Snaptrim_vs_Avg_Clat.png)
+
+<center>Fig. 18: balanced - Avg Client Latency Variations During Recovery+Snaptrim Ops</center><br>
+
+[![](images/Img19_high_recovery_ops_Profile_Rec_vs_Snaptrim_vs_Avg_Clat.png)](images/Img19_high_recovery_ops_Profile_Rec_vs_Snaptrim_vs_Avg_Clat.png)
+
+<center>Fig. 19: high_recovery_ops - Avg Client Latency Variations During Recovery+Snaptrim Ops</center><br>
+
+## Scaled Cluster: Client Ops & Recovery
+
+### Test Environment
+
+The scaled cluster had a total of 975 OSDs (SSD) and 40 hosts with 13 TiB of raw capacity. But for the recovery test, a total of 3 hosts each hosting 25 OSDs was employed. Also, 3 RBD pools (1 pre-existing with ~100 Million 4KiB Objects, 1 pool for client ops and 1 recovery pool with 100K 4 MiB objects) were used for the test. All the Ceph pools were configured with a replication factor of 3. One OSD from each host was restarted to trigger recovery of misplaced objects. Put together, around 650K-730K Objects were recovered during the tests and stats were collected and compared across the mClock profiles. Tests with WPQ scheduler are pending and are not discussed in this blog. The configuration of each node is shown below:
```

Comment:
A sentence to emphasize that this is a logical scale cluster and the intention behind running these tests in this cluster would be great.

Reply:
Sure, I have added a sentence to convey this. Please review.

Reply:
Additionally, in the introduction section at the beginning, I added a sentence to convey that results from testing on a logically scaled cluster is also presented.